src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +152,1305 @@ static DamageTable D_80B65044 = {
+        Math_Vec3f_Diff(&cam->eye, &cam->at, &vec1);
+        magnitude = Math3D_Vec3fMagnitude(&vec1);
+        magnitude2 = (magnitude > 1.0f) ? (20.0f / magnitude) : (20.0f);
+        Math_Vec3f_Scale(&vec1, magnitude2);
+    } else {
+        Math_Vec3f_Copy(&vec1, &D_801D15B0);
+    }
+
+    {
+        GraphicsContext* gfx = globalCtx->state.gfxCtx;
+
+        // fully visible OR fully transparent
+        if ((this->mainColor.a == 255) || (this->mainColor.a == 0)) {
+            Scene_SetRenderModeXlu(globalCtx, 0, 1);
+            dispHead = gfx->polyOpa.p;
+        } else {
+            Scene_SetRenderModeXlu(globalCtx, 1, 2);
+            dispHead = gfx->polyXlu.p;
+        }
+
+        gSPDisplayList(&dispHead[0], &sSetupDL[6 * 0x19]);
+
+        gSPSegment(&dispHead[1], 0x0A, Gfx_EnvColor(globalCtx->state.gfxCtx, 160, 0, 255, this->mainColor.a));
+
+        SysMatrix_GetStateTranslationAndScaledY(1400.0f, &vec2);
+        Lights_PointGlowSetInfo(&this->fires[0].info, vec2.x + vec1.x, vec2.y + vec1.y, vec2.z + vec1.z,
+                                this->lanternColor.r, this->lanternColor.g, this->lanternColor.b, this->lanternColor.a);
+
+        gDPSetEnvColor(&dispHead[2], this->lanternColor.r, this->lanternColor.g, this->lanternColor.b,
+                       this->mainColor.a);
+
+        gSPMatrix(&dispHead[3], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        gSPDisplayList(&dispHead[4], &D_060042C8);
+
+        gSPDisplayList(&dispHead[5], &D_060043F8);
+
+        // fully transparent OR fully invisible
+        if ((this->mainColor.a == 255) || (this->mainColor.a == 0)) {
+            gfx->polyOpa.p = &dispHead[6];
+        } else {
+            gfx->polyXlu.p = &dispHead[6];
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpo/func_80B641E8.s")
+// one of the main draws
+// but also called by another draw function
+void EnBigpo_Draw4(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    s32 pad[3];
+    s16 fireRadius;
+    MtxF* mtfxPtr;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpo/func_80B64240.s")
+    mtfxPtr = SysMatrix_GetCurrentState();
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpo/func_80B64470.s")
+    func_8012C2DC(globalCtx->state.gfxCtx);
+    Matrix_RotateY(BINANG_ROT180(func_800DFCDC(ACTIVE_CAM)), MTXMODE_NEW);
+    if (this->actionFunc == EnBigpo_SpawnPoCutscene6) {
+        Matrix_Scale(0.01f, 0.01f, 0.01f, 1);
+        fireRadius = 0x1F4;
+    } else {
+        Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, 1);
+        fireRadius = (s16)(thisx->scale.x * 500.0f * 100.0f);
+    }
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0,
+                                ((s32)globalCtx->gameplayFrames * -0x14) & 0x1FF, 0x20, 0x80));
```

Comment:
```suggestion
                                ((s32)globalCtx->gameplayFrames * -20) % 512, 0x20, 0x80));
```
If it matches

Reply:
Weird, this one doesn't match...?

![mod 512](https://user-images.githubusercontent.com/42048411/128614766-319169fb-6587-4427-a65f-1c5637f9bac0.png)